src/pages/pass-finalisation/cat-be/components/code-78/code-78.ts

```diff
@@ -0,0 +1,77 @@
+import { Component, Input, OnChanges, Output, EventEmitter } from '@angular/core';
+import { FormGroup, FormControl, Validators } from '@angular/forms';
+import { TestCategory } from '../../../../../shared/models/test-category';
+import { Code78Present, Code78NotPresent } from '../../../../../modules/tests/pass-completion/pass-completion.actions';
+import { GearboxCategory } from '@dvsa/mes-test-schema/categories/Common';
+
+@Component({
+  selector: 'code-78',
+  templateUrl: 'code-78.html',
+})
+
+export class Code78Component implements OnChanges {
+  @Input()
+  form: FormGroup;
+
+  @Input()
+  category: String;
+
+  @Input()
+  transmission: GearboxCategory;
+
+  @Output()
+  code78Present = new EventEmitter<Code78Present>();
+
+  @Output()
+  code78NotPresent = new EventEmitter<Code78NotPresent>();
+
+  formControl: FormControl;
+  static readonly fieldName: string = 'code78Ctrl';
+  manualMessage: string = 'A <b><em>manual</em></b> license will be issued';
+  automaticMessage: string = 'An <b><em>automatic</em></b> license will be issued';
+
+  ngOnChanges(): void {
+    if (!this.formControl) {
+      this.formControl = new FormControl('', [Validators.required]);
+      this.form.addControl(Code78Component.fieldName, this.formControl);
+    }
+  }
+
+  shouldHideBanner(): boolean {
+    return (
+      !this.formControl.valid || !this.transmission
+    );
+  }
+
+  isInvalid(): boolean {
+    return !this.formControl.valid && this.formControl.dirty;
+  }
+
+  shouldShowManualBanner(): boolean {
+    if (!this.shouldHideBanner()) {
+      switch (this.category as TestCategory) {
+        case TestCategory.BE:
+          return this.transmission === 'Manual';
```

Comment:
If you want, you can create a `Transmission` enum too and we can do away with these magic strings